keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -477,11 +477,12 @@ class SqlDeliveryConfigRepository(
               .onDuplicateKeyUpdate()
               .set(CURRENT_CONSTRAINT.CONSTRAINT_UID, MySQLDSL.values(CURRENT_CONSTRAINT.CONSTRAINT_UID))
               .execute()
+          }
 
+          sqlRetry.withRetry(WRITE) {
             val allStates = constraintStateFor(state.deliveryConfigName, state.environmentName, state.artifactVersion)
```

Comment:
If I understand this correctly (and this is a subtle bug!), it looks like `constraintStateFor` executes a SQL query that is outside of the transaction. Is that right?

Reply:
yes, after this change, the transaction is being closed at line 480, and line 482 and after is outside that transaction.

Reply:
I would highly recommend adding a comment here to explain why this is done outside the transaction, as it would be easy to fall into the trap of assuming this was done by mistake and adding this back into the transaction down the road.